web/client-api/src/main/java/io/deephaven/web/client/api/ReconnectState.java

```diff
@@ -72,6 +72,11 @@ public void success() {
      * After the connection closes or attempt fails, should be called.
      */
     public void failed() {
+        if (state == State.Disconnected) {
```

Comment:
This is the 1 change in this PR that actually fixes a bug we are observing.
In [this PR](https://github.com/deephaven-ent/iris/pull/3518) we now explicitly disconnect clients / close connections that have failed auth. Without this check, the clients will just reconnect and continue failing.

The 2 changes in `WorkerConnection` are more theoretical fixes in that there aren't any known issues caused by them, but they seem to be good checks to have.

Reply:
After talking to @niloc132 we decided not to go this route since it will stop explicit `forceReconnect` from working. There seems to be only 1 place that was remaining in `WorkerConnection` where we weren't checking for the explicit disconnect. I've added the change there to see if it fixes our issue.